test/extended/networking/internal_ports.go

```diff
@@ -35,7 +35,7 @@ const (
 var _ = ginkgo.Describe("[sig-network] Internal connectivity", func() {
 	f := framework.NewDefaultFramework("k8s-nettest")
 
-	ginkgo.It("for TCP and UDP on ports 9000-9999 is allowed", func() {
+	ginkgo.It("for TCP and UDP on ports 9000-9999 is allowed [Serial:Self]", func() {
```

Comment:
There's no reason the test has to always use ports 9000 and 9999 exactly... it should just pick two random ports in that range.

Although I guess then you have to worry about conflicting with other components that use ports in that range, so I guess it will need some retrying if the first attempt at pod creation fails. (But then, that makes it more resilient against other copies of itself too.)

Anyway, if not that, I don't think we want the `[Serial:Self]` tag; the `[Serial]` and `[Disruptive]` tags are understood to cover a pretty broad range of incompatible test behaviors, so you can just use `[Serial]` for this case.

Reply:
I agree that if we make it choose random ports, there is no need for this tag, but then we need to add some try-cycle for different port numbers so that the test won't fail if it chooses some service's port. Also finding out if deployment fails because of someone legally using chosen port or not may be difficult and time consuming. So maybe using just edge port values is not a bad idea, what do you think?

But in general, `[Serial:Self]` tests are only excluded from network-stress tests, and basically any test that uses some exclusive resources should be marked like that, but using `[Serial]` label will exclude the test from all parallel test suits (where it passes). And  I think we don't want to exclude tests for no reason.

Also, the problem with running copies of the same test in parallel may be fixed by changing test scheduler behaviour by making it schedule the next copy of the same test only when the previous one is finished, and when this is done, we can remove  `[Serial:Self]` label.